test/tests/controls/pid_control/tests

```diff
@@ -16,7 +16,8 @@
       type = 'CSVDiff'
       input = 'pid_control.i'
       csvdiff = 'min_max_limits.csv'
-      cli_args = 'Outputs/file_base=min_max_limits Controls/integral_value/maximum_output_value=4.0 Controls/integral_value/minimum_output_value=2.0 Executioner/nl_abs_tol=1e-14 Executioner/end_time=10'
+      cli_args = 'Outputs/file_base=min_max_limits Controls/integral_value/maximum_output_value=4.0 Controls/integral_value/minimum_output_value=2.0 
+                  Executioner/nl_abs_tol=1e-14 Executioner/end_time=10 Controls/integral_value/execute_on=timestep_end'
```

Comment:
was that necessary? the regold is due to that right?

Reply:
I can undo this change and add a new test instead. Changing this test so `PIDTransientControl ` executes on timestep end allows this existing test to catch the issue noted in #31842

Reply:
yes let s have another test.
Feel free to just duplicate this one
otherwise it's too hard to verify that the test covers a bunch of independent requirements the next time we modify it